dartagnan/src/main/java/com/dat3m/dartagnan/verification/model/ThreadModel.java

```diff
@@ -0,0 +1,46 @@
+package com.dat3m.dartagnan.verification.model;
+
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.Thread;
+import com.dat3m.dartagnan.verification.model.event.EventModel;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.ArrayList;
+
+
+public class ThreadModel {
+    private final Thread thread;
+    private final List<EventModel> eventList;
```

Comment:
It feel wrong to explicitly keep track of `EventModel`s both in the ``ExecutionModelNext` where the later has access to the `ThreadModel` and could propagate the retrieval of the events

Reply:
I don't think keeping track of `EventModels` inside `ThreadModel` is wrong. It is useful when we want to do something to the `EventModels` which belong to the same `ThreadModel`. And in `ExecutionModelNext` it is also meaningful to have a complete view on all `EventModels`.